### DISCUSSION TITLE
Floating feedback mode Fl_Mode active below rated wind speed

### DISCUSSION THREAD

**Discussion by AntonioPJ23** [NatLabRockies/ROSCO#289] [Q&A]
Hi all

I am using ROSCO v2.7 to simulate the IEA 15MW turbine on a floater. I have enabled the floating feedback mode with a gain value that works fine above rated. However, I see controller-induced floater pitch motion below rated. 

The plots shown are for turbulent wind at 7 m/s. The top panel is the floater pitch and the bottom is blade pitch. The blue curves are obtained with Fl_Mode=0, while the orange corresponds to Fl_Mode=1. As expected, the blade pitch controller without floating feedback (blue) does nothing, since we operate below rated wind speed. But the floating feedback mode (orange) does change the blade pitch, leading to large floater pitch motions.

My question is, why is the floating feedback mode active below rated? Shouldn't this be active above rated only, where the blade pitch PI controller is enabled as well, and where floater pitch instability is likely to happen? Is there a flag somewhere in the input file that I have not seen?

Thanks in advance

![Untitled](https://github.com/NREL/ROSCO/assets/152971470/907d46e4-1304-4e07-b02f-4bfbf03e6c8c)

**Comment by dzalkind** [comment]
Hi, in ROSCO 2.7.0, floating feedback is always enabled, regardless of the wind speed.  In the `develop` branch and future 2.9.0, you can [gain-schedule the floating feedback](https://github.com/NREL/ROSCO/blob/c75738eee4ab4864e1c9d344ce2cfc0dd80f272d/Test_Cases/IEA-15-240-RWT-UMaineSemi/DISCON-UMaineSemi.IN#L140).  

[A script to update discons can be found here](https://github.com/NREL/ROSCO/blob/develop/Examples/19_update_discon_version.py).  You will need to [compile ROSCO from source](https://rosco.readthedocs.io/en/latest/source/install.html#compile-using-cmake) to use this feature until it is released in 2.9.0.  [This example](https://github.com/NREL/ROSCO/blob/develop/Examples/24_floating_feedback.py) may also be of interest.

Do you mean to have the minimum pitch limit of -3 deg.?  The label also says it's for the IEA-15MW, so I'm guessing this was a deliberate change.  When the pitch increases, I'm guessing the thrust increases, which is the opposite of what usually happens when the fine pitch/min pich is greater than or equal to 0, leading to this behavior.  In usual cases, some floating feedback, which is usually small, is not an issue for this turbine model.

I hope this helps.

Best, Dan

**Reply by AntonioPJ23** [reply]
Hi Dan

Thank you very much for your quick reply. It is good to know that gain-scheduling for the floating feedback loop will be included in the new version - I am looking forward to it :-)

Good point about the minimum blade pitch. This has been indeed deliberately changed, although not by me. I have repeated the same simulation using the default IEA15MW minimum pitch settings and the instability in floater pitch still happens, although with smaller amplitude (see below, with new results in green). 

I am not sure that having a negative mean blade pitch will necessarily lead to a change in sign in the derivative of thrust with blade pitch (and hence negative aerodynamic damping). I guess it requires that a good portion of the blade operates in the stall region, where a reduction in angle of attack causes an increase in the lift coefficient.

Thanks again for your help!
Best,
Antonio

![image](https://github.com/NREL/ROSCO/assets/152971470/2a8b6411-cdd9-4224-8422-7c43c9fbd104)

**Reply by dzalkind** [reply]
I'm guessing you also changed the platform of the turbine since it has a negative mean pith angle.  To remove the instability, you can try reducing the `omega_pc` of the pitch controller for speed.  We're also learning that `ptfm_freq`, which becomes `F_FlCornerFreq`, does have and effect on the damping and stability of some platforms.
